.github/workflows/main.yml

```diff
@@ -0,0 +1,109 @@
+name: UX
+on: [push]
+
+jobs:
+  test_keychain:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Set Node Version
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.16.1
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - uses: actions/cache@v1
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    - name: Install monorepo deps
+      run: yarn
+    - name: Install Keychain dependencies
+      run: yarn
+      working-directory: ./packages/keychain
+    - name: Keychain tests
+      run: yarn lerna run test --scope @blockstack/keychain
+
+  codecheck:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Set Node Version
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.16.1
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+
+    - uses: actions/cache@v1
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    - name: Monorepo Dependencies
+      run: yarn --frozen-lockfile
+    - name: Lerna Bootstrap
+      run: yarn lerna bootstrap
+    - name: Lint
+      run: yarn lint
+    - name: Depcheck
+      run: yarn lerna run depcheck --parallel
+    - name: Typecheck
+      run: yarn lerna run typecheck --parallel
+
+  publish:
+    needs: [test_keychain, codecheck]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Checkout master
+      run: git checkout master
```

Comment:
For some reason, `lerna version` fails if I don't add this. It throws an error about being in a detached Git state.

Reply:
Is this because the default branch is develop?

Reply:
No - I'm not sure exactly how `actions/checkout` works - it definitely checks out the same exact code as whatever the action is running on, but it's in a detached HEAD, even when I add `ref: master` to the job.

There is probably something else I could do to make this work, but for now, this seems to work just fine.